ntbs-service/Pages/Notifications/Edit/Patient.cshtml.cs

```diff
@@ -25,27 +26,33 @@ public class PatientModel : NotificationEditModelBase
         [BindProperty]
         [ValidFormattedDateCanConvertToDatetime(ErrorMessage = ValidationMessages.InvalidDate)]
         public FormattedDate FormattedDob { get; set; }
-        private IPostcodeService PostcodeService { get; set; }
+        private readonly IPostcodeService postcodeService;
 
-        public PatientModel(INotificationService service, IPostcodeService postcodeService, NtbsContext context) : base(service)
+        public PatientModel(INotificationService service, IPostcodeService postcodeService, IAuthorizationService authorizationService, NtbsContext context) : base(service, authorizationService)
         {
-            PostcodeService = postcodeService;
+            this.postcodeService = postcodeService;
             Ethnicities = new SelectList(context.GetAllEthnicitiesAsync().Result, nameof(Ethnicity.EthnicityId), nameof(Ethnicity.Label));
             Countries = new SelectList(context.GetAllCountriesAsync().Result, nameof(Country.CountryId), nameof(Country.Name));
             Sexes = context.GetAllSexesAsync().Result.ToList();
         }
 
         public override async Task<IActionResult> OnGetAsync(int id, bool isBeingSubmitted)
         {
-            Notification = await service.GetNotificationAsync(id);
-            if (Notification == null)
```

Comment:
The actual logic is now happening in the Method "SetNotificationAndAuthorize", but there will still be some repetition in handling the various exceptions/page routes across all the page models.

The only alternative I can see is to put this into base.OnGetAsync, which could then return either null or an action result - the parent OnGetAsync method could then do
```
var actionResult = base.OnGetAsync(id);
if (actionResult != null)
{
   return actionResult
}
```

Do you think this would be worth it?

Reply:
Or, we can follow what we've done for POST, where:
- incoming POST is picked up by `OnPostAsync` on the base
- the base calls `ValidateAndSave` on the children (and imposes they implement it by being a virtual method)

Reply:
given the amount of places we're running the same change in atm, I think some commonisation like this would be worth it!

Reply:
@jasiekmiko I feel like we shouldn't need to define OnGetAsync here, and instead it should just call OnGetAsync on the base method. However for some reason if I remove this the base method gets called with notification id= 0 - I think the handler isn't quite setup correctly, but not sure how to fix it -do you know by any chance?

If not I'll dig into it, but thought it would be good to get this merged first. I will add the tests after merging, as this is really far behind master and things have moved on quite a bit since in the integration tests!

Reply:
OK, good to know what's happening there.
Not sure of the solution off the top of my head, but as @conorsheehansw has been adjusting the paths atm too, so let's not get it too messy on two different streams.
Unless @conorsheehansw you have any idea straight away?

Reply:
I would've thought that an OnGetAsync in the base would be sufficient, but it not having binding seems strange - would've thought it would have access to the same context...

Something like this might be the route to being able to commonise parts of it with it being quite still quite traceable:
https://www.learnrazorpages.com/razor-pages/filters#creating-a-filter-attribute

Reply:
Small amount of exploration here, had the base OnGetAsync correctly binding to the route (though this was in parallel with my custom routing via `@page "/Notifications/{id?}/{handler?}"`

Reply:
@conorsheehansw So it works on your branch? Could I leave it to you to remove the unnecessary OnGets in the individual models then? :)

Reply:
Just merged from master into my branch, to confirm it would work... 
Reason why it wasn't working in your testing, was that in the method declaration for base `OnGet` it was requesting for `notificationId` not `id` as specified in query string(/and in the new routing declarations).
Yeah I can handle stripping the proxy OnGets from the pagemodels :).

Reply:
Ahh nice that makes sense - and that also shows that routing with query strings is way more likely to break